### PR TITLE
feat: theme 2 menu with headers, subheaders, footers and links

### PR DESCRIPTION
**Pull request by ychhabra-eightfold**
## SUMMARY:
feat: theme 2 menu with headers, subheaders, footers and links

## JIRA TASK (Eightfold Employees Only):
https://eightfoldai.atlassian.net/browse/ENG-23860

## CHANGE TYPE:

-   [ ] Bugfix Pull Request
-   [ ] Feature Pull Request

## TEST COVERAGE:

-   [ ] Tests for this change already exist
-   [ ] I have added unittests for this change

### CODE REVIEW

File: src/components/Dropdown/dropdown.module.scss [hunk 1]
```diff
@@ -28,7 +28,7 @@
     opacity: 0;
     white-space: normal;
     z-index: $z-index-400;
-    max-height: 34vh;
+    max-height: min(50vh, 600px);
```
Comment (jhoward-eightfold) [inline src/components/Dropdown/dropdown.module.scss:31]: min() is not supported on IE11. Is "graceful" degradation full viewport height?

File: src/components/Menu/MenuItem/MenuItemSubHeader/MenuItemSubHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import styles from '../menuItem.module.scss';
+
+export const MenuItemSubHeader: FC<MenuItemSubHeaderProps> = ({
+    text,
+    size,
+}) => {
+    const subHeaderClasses: string = mergeClasses([
+        styles.menuItemSubHeader,
+        { 'body3 semibold': size === MenuSize.small },
```
Comment (jhoward-eightfold) [inline src/components/Menu/MenuItem/MenuItemSubHeader/MenuItemSubHeader.tsx:14]: would prefer we course correct on number based sizing classes. we should call small "sm", large "lg", etc. the numbers aren't intuitive.

Comment (jhoward-eightfold) [reply src/components/Menu/MenuItem/MenuItemSubHeader/MenuItemSubHeader.tsx:14]: Or single letter sizing ($space-s), as seem below in this PR, but we should establish a standard way of denoting size.

Comment (ychhabra-eightfold) [reply src/components/Menu/MenuItem/MenuItemSubHeader/MenuItemSubHeader.tsx:14]: MenuSize.small is "small", not using numbers here

Comment (jhoward-eightfold) [reply src/components/Menu/MenuItem/MenuItemSubHeader/MenuItemSubHeader.tsx:14]: sorry, was referring to the body3. More of a comment for near future consideration than something I'm looking for action on in this PR

**Review by dkilgore-eightfold — APPROVED** [review]

**Pull request merged**